tests/unit/extended_datastore_adapter/test_get.py

```diff
@@ -101,3 +104,14 @@ def test_get_only_db_exception(self) -> None:
             )
         self.db_method_mock.assert_called()
         self.add_get_many_mock.assert_not_called()
+
+    def test_get_empty_mapped_fields_and_changed_models(self) -> None:
```

Comment:
IMO this test only tests the mock, but not the changed _get_many_from_changed_models-method. From code-review I think the change is okay, but ...;-)

Reply:
No, I wrote the test before I changed the code and it failed. The changed method is not mocked here, only the datastore - the fake mok datastore used here had to be changed because it did not know about the "empty mapped-fields" thing ;)

Reply:
You are right, the changed method is called via mock

Reply:
But there are 2 more things in the get-method itself.
1. Performance issue: Should have a flag that the whole model is read into change_models "cache" to prevent reading the whole model each time, when there are no mapped fields
2. The get method has only 1 fqid and can only work with 1 fqid, but the method pretends to be usable with more fqids. Shouldn't we implement it without fqid-dict or made it usable for more than 1 fqid?

Reply:
> 1. Performance issue: Should have a flag that the whole model is read into change_models "cache" to prevent reading the whole model each time, when there are no mapped fields

I don't think so, this is a task for the real cache - see #1289.

> 2. The get method has only 1 fqid and can only work with 1 fqid, but the method pretends to be usable with more fqids. Shouldn't we implement it without fqid-dict or made it usable for more than 1 fqid?

I don't understand what you mean. What get method? Both the one in `extended_datastore_adapter.py` as well as the `_get_mock` in this file only take one fqid.

Reply:
> > 1. Performance issue: Should have a flag that the whole model is read into change_models "cache" to prevent reading the whole model each time, when there are no mapped fields
> 
> I don't think so, this is a task for the real cache - see #1289.
> 
> > 2. The get method has only 1 fqid and can only work with 1 fqid, but the method pretends to be usable with more fqids. Shouldn't we implement it without fqid-dict or made it usable for more than 1 fqid?
> 
> I don't understand what you mean. What get method? Both the one in `extended_datastore_adapter.py` as well as the `_get_mock` in this file only take one fqid.

The adapter get-method: With the missing_fields_per_fqid-dict it seems to be made for more than 1 fqid, but it in't. We could use a simple fields-list instead a dict with list.